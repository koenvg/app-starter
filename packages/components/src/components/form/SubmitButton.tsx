import { Button } from "../button";
import React, { SFC } from "react";
import { useFormikContext } from "formik";

export const SubmitButton: SFC = ({ children }) => {
  const { handleSubmit, isSubmitting } = useFormikContext();
  return (
    <Button type="submit" color="primary" variant="contained" onClick={handleSubmit as any} loading={isSubmitting}>
      {children}
    </Button>
  );
};

SubmitButton.displayName = "SubmitButton";
