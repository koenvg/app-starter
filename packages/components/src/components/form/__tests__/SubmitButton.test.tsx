import { SubmitButton } from "../SubmitButton";
import React from "react";
import { render, fireEvent, wait } from "@testing-library/react";
import { Formik, Form } from "formik";

describe("SubmitButton", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  const submitFn = jest.fn();
  const SimpleForm = () => {
    return (
      <Formik onSubmit={submitFn} initialValues={{ name: "" }}>
        <Form>
          <SubmitButton>Submit</SubmitButton>
        </Form>
      </Formik>
    );
  };

  it("Should render a form with the submitButton", () => {
    const { container } = render(<SimpleForm />);
    expect(container).toMatchSnapshot();
  });

  it("should submit the form when you click the button and show a loading indicator", async () => {
    const { container } = render(<SimpleForm />);
    const button = container.querySelector("button");
    await wait(() => fireEvent.click(button));
    expect(submitFn).toHaveBeenCalled();
    expect(container).toMatchSnapshot();
  });
});
