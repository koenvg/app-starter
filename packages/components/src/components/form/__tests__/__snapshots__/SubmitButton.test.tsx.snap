// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubmitButton Should render a form with the submitButton 1`] = `
<div>
  <form
    action="#"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
      tabindex="0"
      type="submit"
    >
      <span
        class="MuiButton-label"
      >
        Submit
      </span>
    </button>
  </form>
</div>
`;

exports[`SubmitButton should submit the form when you click the button and show a loading indicator 1`] = `
<div>
  <form
    action="#"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary Mui-disabled Mui-disabled"
      disabled=""
      tabindex="-1"
      type="submit"
    >
      <span
        class="MuiButton-label"
      >
        <div
          class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
          role="progressbar"
          style="width: 25px; height: 25px;"
        >
          <svg
            class="MuiCircularProgress-svg"
            viewBox="22 22 44 44"
          >
            <circle
              class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
              cx="44"
              cy="44"
              fill="none"
              r="20.2"
              stroke-width="3.6"
            />
          </svg>
        </div>
      </span>
    </button>
  </form>
</div>
`;
